### DISCUSSION TITLE
Publishing a Slice of Data as Multiple Simple Readings in EdgeX Sink

### DISCUSSION THREAD

**Discussion by younes199511** [lf-edge/ekuiper#2854] [Q&A]
Hello,

I'm working on a project where I have a slice of data that I need to publish in the EdgeX sink. My goal is to have each item in the slice published as a separate simple reading instead of array type, either within a single event or across multiple events.

However, I'm currently facing a challenge. When I try to publish the data, the EdgeX sink always combines the data into a single reading, even when I use a [data template](https://ekuiper.org/docs/en/latest/guide/sinks/data_template.html) or [unnest ](https://ekuiper.org/docs/en/latest/sqls/functions/multi_row_functions.html)function. I was wondering if there's a way in eKuiper to publish each item in the slice as a separate reading, rather than having them all in one reading.

Below a simple test rule where i want to publish the items of the slice in separate readings (one or multiple events)
```
{
  "id": "ruleRedisEvent",
  "sql": "SELECT unnest(array_create(10,15,35)) AS data FROM events",
  "actions": [
    {
      "edgex": {
        "protocol": "redis",
        "host": "localhost",
        "port": 6379,
        "topic": "application",
        "profileName": "ekuiperProfile",
        "deviceName": "ekuiper",      
        "contentType": "application/json"
      }
    }
  ]
}
```

Any guidance or suggestions would be greatly appreciated.

Thank you

**Comment by ngjaying** [comment]
Add sendSingle: true to the edgex action. Could you have a try?

**Reply by younes199511** [reply]
Thank you for your response. I tried it, but it didn't work as expected (it's still sending a float64Array).
![image](https://github.com/lf-edge/ekuiper/assets/44032734/4632ea76-a520-4c4e-8ff5-09b7cb21b439)

I attempted to write a data template to separate the items of the slice as follows (edgex sink sends one event with readings comp_result0 and comp_result1)
`{\"comp_result0\":{{index .comp_result 0}},\"comp_result1\":{{index .comp_result 1}}}`

but I'm still encountering an issue. I couldn't make it generic despite trying multiple approaches (the edgex sink isn't sending anything, with no errors displayed in the logs). Additionally, I couldn't find a way to publish all the items as readings of the same resourceName.
